src/components/hooks/useBookmarks.js

```diff
@@ -1,59 +1,154 @@
 import useSWR from "swr";
 import * as workFragments from "@/lib/api/work.fragments";
-import { useData } from "@/lib/api/api";
+import { useData, useMutate } from "@/lib/api/api";
+import { useEffect } from "react";
+import useUser from "./useUser";
+import * as bookmarkMutations from "@/lib/api/bookmarks.mutations";
+import * as bookmarkFragments from "@/lib/api/bookmarks.fragments";
+import { useSession } from "next-auth/react";
 
 const KEY_NAME = "bookmarks";
 
+export const BookmarkSyncProvider = () => {
+  const { syncCookieBookmarks } = useBookmarks();
+  const { data: session } = useSession();
+
+  useEffect(() => {
+    const isAuthenticated = !!session?.user?.uniqueId;
+    if (isAuthenticated) {
+      syncCookieBookmarks().then(() => {});
+    }
+  }, [session]);
+
+  return <></>;
+};
+
 export default function useBookmarks() {
+  const { data: session } = useSession();
+  const isAuthenticated = !!session?.user?.uniqueId;
+
   let {
-    data: bookmarks,
-    mutate: mutateBookmark,
+    data: localBookmarks,
+    mutate: mutateLocalBookmarks,
     error,
   } = useSWR(KEY_NAME, (key) => JSON.parse(localStorage.getItem(key)) || []);
+  const {
+    data: globalBookmarksUserObject,
+    isLoading: isLoadingGlobalBookmarks,
+    error: globalBookmarksError,
+    mutate: mutateGlobalBookmarks,
+  } = useData(isAuthenticated && bookmarkFragments.fetchAll());
+  const bookmarkMutation = useMutate();
+  const { isLoggedIn } = useUser();
+  const globalBookmarks =
+    globalBookmarksUserObject?.user?.bookmarks?.result?.map((bookmark) => ({
+      ...bookmark,
+      key: bookmark.materialId + bookmark.materialType,
+    }));
+
+  const syncCookieBookmarks = async () => {
+    if (!isAuthenticated) return; // Not authenticated
+    const cookies = await JSON.parse(localStorage.getItem(KEY_NAME));
+    if (!cookies || !Array.isArray(cookies) || cookies.length === 0) return; // Nothing to sync
+
+    try {
+      await bookmarkMutation.post(
+        bookmarkMutations.addBookmarks({
+          bookmarks: cookies.map((bookmark) => ({
+            materialId: bookmark.materialId,
+            materialType: bookmark.materialType,
+          })),
+        })
+      );
+
+      await mutateGlobalBookmarks();
+
```

Comment:
Vil du have mulighed for at sende titlen med? Det skal vi bruge til at sortere alfabetisk.

Reply:
Se bort fra min kommentar😬 FBI-api skal rettes først